node/src/reactor/joiner.rs

```diff
@@ -508,12 +509,12 @@ impl reactor::Reactor for Reactor {
 
         let (consensus, init_consensus_effects) = EraSupervisor::new(
             timestamp,
-            EraId(0),
+            chainspec_loader.initial_era(),
             WithDir::new(root, config.consensus.clone()),
             effect_builder,
             validator_weights,
             chainspec_loader.chainspec().as_ref().into(),
-            chainspec_loader.starting_state_root_hash(),
+            chainspec_loader.initial_state_root_hash(),
```

Comment:
It looks like we actually need the _genesis_ state root hash here: I think it's only used to compute the instance ID of era 0?

Reply:
I think currently you're right... I was thinking that this would be used to get the validator sets via the contract runtime, but it isn't, so we should probably go back to passing the genesis state root hash here for now.

Reply:
If we pass in the Genesis state will it work with upgrades?

Reply:
It _will_ actually be used to get the validator set via the contract runtime — but only for era 0! (I'll open a PR for that as soon as this one is merged; but we'll need to pass in the genesis root hash, yes.)

Reply:
But I think even before this PR it was wrong, and the `starting_state_root_hash` wouldn't be the genesis root hash if we _restarted_ shortly after genesis?

Reply:
> If we pass in the Genesis state will it work with upgrades?

No, for upgrades we'll need the protocol state from right after the upgrade (i.e. with changes applied), which we also don't have yet.

Reply:
OK, that was my point. So it sounds to me like `chainspec_loader.initial_state_root_hash()` is a good method to call but it should be fixed what it returns?

Reply:
We'll need additional functionality to fix that. I ticketed it as NDRS-943 and will do it as a follow-up to this PR.